pkg/vpcmodel/grouping.go

```diff
@@ -219,24 +219,44 @@ func (g *groupingConnections) addPublicConnectivity(ep EndpointElem, commonProps
+// vsiOrSubnetsGroupingBySubnetsOrVsis given *GroupConnLines, a list of EndpointElem and a bool saying whether
+// the EndpointElem represents VSIs or subnets.
+// It returns a slice of EndpointElem objects, by grouping the input set of EndpointElem into the input *GroupConnLines
+// such that
+// 1. If the grouped elements are vsis (as by the input bool) then the elements are grouped by their subnet
+// 2. If the grouped elements are subnets then the elements are grouped by their VPC, this is automatic
+// unless VPCConfig is result of a dummy vpc built for tgw, namely IsMultipleVPCsConfig = true
+func vsiOrSubnetsGroupingBySubnetsOrVsis(groupedConnLines *GroupConnLines,
+	elemsList []EndpointElem, groupVSI bool) []EndpointElem {
 	res := []EndpointElem{}
-	subnetNameToNodes := map[string][]EndpointElem{} // map from subnet name to its nodes from the input
+	// map from subnet's/vsi's UID to its nodes/nodeSets from the input
+	subnetOrVSIToNodesOrNodeSets := map[string][]EndpointElem{}
 	for _, elem := range elemsList {
-		n, ok := elem.(InternalNodeIntf)
-		if !ok {
-			res = append(res, elem) // elements which are not interface nodes remain in the result as in the original input
-			continue                // skip input elements which are not a network interface node
+		var subnetOrVSIUID string
+		var newElem EndpointElem
+		if groupVSI {
+			n, ok := elem.(InternalNodeIntf)
+			if !ok {
+				res = append(res, elem) // elements which are not interface nodes remain in the result as in the original input
+				continue                // skip input elements which are not a network interface node
+			}
+			subnetOrVSIUID = n.Subnet().UID() // get the subnet to which n belongs
+			newElem = elem
+		} else {
+			n, ok := elem.(Subnet)
+			if !ok {
+				res = append(res, n) // elements which are not subnets remain in the result as in the original input
+				continue             // skip input elements which are not a subnet nodeSet
+			}
+			subnetOrVSIUID = n.VPC().UID() // get the VPC to which n belongs
+			newElem = n
 		}
-		subnetName := n.Subnet().Name() // get the subnet to which n belongs
-		if _, ok := subnetNameToNodes[subnetName]; !ok {
-			subnetNameToNodes[subnetName] = []EndpointElem{}
+		if _, ok := subnetOrVSIToNodesOrNodeSets[subnetOrVSIUID]; !ok {
+			subnetOrVSIToNodesOrNodeSets[subnetOrVSIUID] = []EndpointElem{}
 		}
-		subnetNameToNodes[subnetName] = append(subnetNameToNodes[subnetName], elem)
+		subnetOrVSIToNodesOrNodeSets[subnetOrVSIUID] = append(subnetOrVSIToNodesOrNodeSets[subnetOrVSIUID], newElem)
 	}
-	for _, nodesList := range subnetNameToNodes {
+	for _, nodesList := range subnetOrVSIToNodesOrNodeSets {
 		if len(nodesList) == 1 { // a single network interface on subnet is just added to the result (no grouping)
 			res = append(res, nodesList[0])
 		} else { // a set of network interfaces from the same subnet is grouped by groupedNetworkInterfaces object
```

Comment:
this comment (by the `else`) also seems outdated

Reply:
fixed